include/lemlib/chassis/differential/motions/motion.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "lemlib/chassis/differential/drivetrain.hpp"
+
+namespace lemlib::differential {
+/**
+ * @brief abstract motion class
+ *
+ * This is an abstract motion class for a differential drivetrain. It can be used to create
+ * custom motions for the chassis, implemented as part of LemLib, part of an extension for
+ * LemLib, or a user made extension specific to a single project.
+ *
+ * @b Example Child Class
+ * @code {.cpp}
+ * class SimpleForward : public Motion {
+ *   public:
+ *     SimpleForward() {}
+ *
+ *     void calculate(Drivetrain& drivetrain, const ChassisState& state) override {
+ *       drivetrain.left.move(127);
+ *       drivetrain.right.move(127);
+ *     }
+ *
+ *     bool isFinished() override {
+ *       return false; // this motion never finishes
+ *     }
+ * };
+ * @endcode
+ */
+class Motion {
+    public:
+        /**
+         * @brief calculate and apply new outputs for the drivetrain
+         *
+         * @param drivetrain a reference to the drivetrain. This is passed here instead of the ctor to
+         * simplify the creation of the motion, and to discourage the use of the drivetrain outside of
+         * the calculate method
```

Comment:
What does the drivetrain actually do? eg: it provides calculate method the ability to manipulate the drivetrain's motors